### PR TITLE
Reorganize Settings Menu

### PR DESCRIPTION
**Pull request by phillipm**
__Release Notes__: Reorganize settings menu

- Reorganize setting menu entries
- Add confirmation dialog to "Clear User Data"
- Add "Advanced" menu for launching advanced functionality
- Move server settings into own menu

| Before     | After |
| ---      | ---       |
| screen | screen   |
|   screen    |screen       |

| New 'Server Settings' Screen     | New 'Advanced' Screen |
| ---      | ---       |
|    screen  |   screen  |

screen

### CODE REVIEW

**Comment by phillipm** [comment]
cc @dimagi/product

**Comment by amsagoff** [comment]
:heart_decoration:

**Comment by amstone326** [comment]
Was there any specific logic that went into the ordering of the items on the 3 preference screens (Application Preferences, Serving Settings, and Advanced)? If not, there might be something to be said for just alphabetizing?..

File: app/src/org/commcare/preferences/CommCarePreferences.java [hunk 1]
```diff
@@ -39,15 +39,26 @@
...
-    // So these are stored in the R files, but I don't seem to be able to figure out how to pull
-    // them out cleanly?
+    private final static String TAG = CommCarePreferences.class.getSimpleName();
+
+    /**
+     * Entries used as buttons; aren't actually stored preferences
+     */
+    private final static String SERVER_SETTINGS = "server-settings";
+    private final static String DEVELOPER_SETTINGS = "developer-settings";
+    private final static String CLEAR_SAVED_SESSION = "clear-saved-session";
```
Comment (amstone326) [inline app/src/org/commcare/preferences/CommCarePreferences.java]: Should this maybe be in the Advanced Actions activity?

**Comment by phillipm** [comment]
Ordering is based on usage metrics. I did update the order of the home screen drop down a bit just now to put "Settings" and "Advanced" last so they are easier to pick out.

File: app/src/org/commcare/preferences/CommCarePreferences.java [hunk 2]
```diff
@@ -79,85 +105,137 @@
...
+                    String localizedString = Localization.get(prefToTitleMap.get(key));
+                    screen.getPreference(i).setTitle(localizedString);
+                } catch (NoLocalizedTextException nle) {
+                    Log.w(TAG, "Unable to localize: " + prefToTitleMap.get(key));
+                }
+            }
+        }
+    }
+
+    public static void addBackButtonToActionBar(Activity activity) {
```
Comment (amstone326) [inline app/src/org/commcare/preferences/CommCarePreferences.java]: Why is this needed now?

Comment (phillipm) [inline app/src/org/commcare/preferences/CommCarePreferences.java]: Just and improvement that doesn't have to do with other code changes. Adds a `<` to the action bar nav. If the back button on a device is broken (has happened to me before) then getting out of menus without the `<` is near impossible.

File: unit-tests/src/org/commcare/android/tests/caselist/EntityListCalloutDataTest.java [hunk 3]
```diff
@@ -75,6 +75,7 @@ public void testAttachCalloutResultToListTest() {
 
         // ensure that the entity list is filtered by the received callout
         // result data (fingerprint identification list with confidence score)
+        loadList();
```
Comment (amstone326) [inline unit-tests/src/org/commcare/android/tests/caselist/EntityListCalloutDataTest.java]: The changes in this file look like they possibly don't belong in this PR?

Comment (phillipm) [inline unit-tests/src/org/commcare/android/tests/caselist/EntityListCalloutDataTest.java]: yeah, happy to create a new PR if you would like. I just noticed random failing tests and wanted to try fixing them.

File: app/src/org/commcare/preferences/CommCarePreferences.java (hunk 2 above)
Comment (amstone326) [inline app/src/org/commcare/preferences/CommCarePreferences.java]: Ah ok got it. I just noticed while playing around that the one on the Server Settings screen isn't working.

File: unit-tests/src/org/commcare/android/tests/caselist/EntityListCalloutDataTest.java (hunk 3 above)
Comment (amstone326) [inline unit-tests/src/org/commcare/android/tests/caselist/EntityListCalloutDataTest.java]: Got it, like failing in this PR or just in general? Seems fine either way, didn't mean to nitpick, just wasn't sure where it came from/if it was intentional.

**Comment by amstone326** [comment]
@phillipm Awesome, looks really nice. Good to merge once the action bar back button on the Server Settings screen is fixed

**Comment by phillipm** [comment]
Thanks for the review (the diff was pretty rough, sorry..) and catching that bug. I've updated the PR with a fix
